packages/back-end/src/controllers/metrics.ts

```diff
@@ -296,10 +290,11 @@ export async function getMetricsFromTrackedEvents(
     return;
   }
 
-  req.checkPermissions(
-    "createMetrics",
-    dataSourceObj.projects?.length ? dataSourceObj.projects : ""
-  );
+  // When we create auto metrics, they inherit the data source's projects, so we check if the user
+  // has permission to createMetrics for the data source's projects
+  if (!context.permissions.canCreateMetric(dataSourceObj)) {
```

Comment:
Change this so it's clearer that we're passing in a metric-to-be-created here and not a data source:

```ts
if (!context.permissions.canCreateMetric({ projects: dataSourceObj.projects })) {
```